sdk/storage/Azure.Storage.Blobs/src/PartitionedDownloader.cs

```diff
@@ -69,7 +69,7 @@ internal class PartitionedDownloader
             if (transferOptions.InitialTransferSize.HasValue
                 && transferOptions.InitialTransferSize.Value > 0)
             {
-                _initialRangeSize = transferOptions.MaximumTransferSize.Value;
+                _initialRangeSize = Math.Min(transferOptions.InitialTransferSize.Value, _rangeSize);
```

Comment:
why did this change from constant to `_rangeSize` ?

Reply:
I noticed from the test DownloadsInBlocksWhenOverTheLimit (Just above the one I added) for a file of 100 bytes that when the option MaximumTransferSize is specified to 10, and initial to 20, it expects to send 10 requests. It seemed to imply that MaximumTransferSize should cap the InitialTransferSize value so I reflected it in the change.
Not sure if we truly want to enforce that, since the user provides both values we might want to keep it as I put originally?

Reply:
I think the expectation in that test is rather outcome of this bug. I.e. the expected value has been set as a result of running it rather than consciously chosen.
I went through revision history and this seems to be original version before bug was introduced, see https://github.com/Azure/azure-sdk-for-net/blob/117104e440e65f59adf7e11ac5f5ff4cb2377f3d/sdk/storage/Azure.Storage.Blobs/src/PartitionedDownloader.cs#L40-L43 .
I suggest to bring back logic you had in original version and adjust DownloadsInBlocksWhenOverTheLimit test.

Reply:
Giving it more thoughts I think we shouldn't cap this parameter. See description https://github.com/Azure/azure-sdk-for-net/blob/781e1f79a9e774ba582d987a59ad55d8e7b9acfa/sdk/storage/Azure.Storage.Common/src/StorageTransferOptions.cs#L50-L55 . I.e. we shouldn't block anybody from setting it high if they want to download content in one shot.

The limit imposed on MaximumTransferSize makes sense as we start parallelizing download when we cross InitialTransferSize. I suggest to change it to `_initialRangeSize = transferOptions.InitialTransferSize.Value`

Reply:
Sounds good. I did the change and adjusted the old test to match.